Sources/Rendering/OpenGL/ImageMapper/index.js

```diff
@@ -94,6 +104,11 @@ function vtkOpenGLImageMapper(publicAPI, model) {
     shaders.Geometry = '';
   };
 
+  publicAPI.replaceShaderCoincidentOffset = createReplaceShaderCoincidentOffset(
+    publicAPI,
+    model
+  );
```

Comment:
Sorry for keep changing my mind but I don't necessary have a good picture of what you are doing but what about doing something like assuming the binding of the functions to publicAPI should always be the same ones and so on... What do you think?

```
export function extend(publicAPI, model, initialValues = {}) {
  Object.assign(model, DEFAULT_VALUES, initialValues);
  // Inheritance
  vtkViewNode.extend(publicAPI, model, initialValues);
  vtkReplacementShaderMapper.implementReplaceShaderCoincidentOffset(publicAPI, model, initialValues);
  [...]
}
```

Reply:
Is this new change what you had in mind ?

Reply:
Basically when you start providing `(publicAPI, model)` to a function to generate something it really feels like our `extend` methods and therefore it could be used that way rather than rewiring methods outside of that call. But again, I don't have a good picture of what you are doing. 
So I don't know, I'm just making suggestion here on how that code could be refactored.

So far, your change is "good enough" and does not require anything else.

Reply:
Well we're basically trying to factorize a single method between Image and Polydata mappers to avoid duplicates.

As it's only one function, it would be a bit heavy handed to create a parent class, so indeed the current way of adding this function to `publicAPI` during `extend` seems like a good compromise.

Reply:
This is technically what you already did, not a new class hierarchy. But since you already provide `publicAPI` there is no reason that you can not do the following:

```
function implementReplaceShaderCoincidentOffset(publicAPI, model) {
   publicAPI.replaceShaderCoincidentOffset = (a,b,c) => {};
}
++
implementReplaceShaderCoincidentOffset(publicAPI, model);
```

Rather than doing

```
function createReplaceShaderCoincidentOffset(publicAPI, model) {
   return (a,b,c) => {};
}

++
publicAPI.replaceShaderCoincidentOffset = createReplaceShaderCoincidentOffset(
    publicAPI,
    model
  );
```

This is especially true if the methods name is always the same and therefore no typo can be made...

Reply:
I completely agree. I changed the implementation to apply this suggestion in the latest change.

Reply:
This is interesting, thanks @jourdain .

Now the question is, should the same mechanism be done for exposing "resolve coincident 
 topology" parameters to vtkImageMapper?

Current implementation changes the superclass of vtkImageMapper to be vtkMapper instead of vtkAbstractMapper but maybe that is not needed or worst, that is not wanted. The difference between vtkAbstractMapper, vtkAbstractMapper3D and vtkMapper is not clear to me.

Reply:
@martinken might be able to shade some light. 

Otherwise with vtk-js I tried to enable as much composition as possible to prevent writing too much code and reduce code mistake.
 
The macro.js is indeed providing the common method generator, but other class could start gathering several `implementation` rather than having only one like you have in your `ReplacementShaderMapper`.